elide-core/src/main/java/com/yahoo/elide/core/pagination/Pagination.java

```diff
@@ -19,30 +19,33 @@
  * Encapsulates the pagination strategy.
  */
 
-@AllArgsConstructor
 @ToString
 public class Pagination {
 
     /**
      * Denotes the internal field names for paging.
      */
-    public enum PaginationKey { offset, limit, page }
+    public enum PaginationKey { offset, limit, page, totals }
 
     public static final int DEFAULT_PAGE_SIZE = 500;
     public static final int MAX_PAGE_SIZE = 10000;
 
-    private static final Pagination DEFAULT_PAGINATION = new Pagination(0, 0, false);
+    private static final Pagination DEFAULT_PAGINATION = new Pagination(new HashMap<>());
+
+    // For requesting total pages/records be included in the response
+    public static String PAGE_TOTALS_KEY = "page[totals]";
 
     public static final Map<String, PaginationKey> PAGE_KEYS = new HashMap<>();
     static {
         PAGE_KEYS.put("page[size]", PaginationKey.limit);
         PAGE_KEYS.put("page[limit]", PaginationKey.limit);
         PAGE_KEYS.put("page[number]", PaginationKey.page);
         PAGE_KEYS.put("page[offset]", PaginationKey.offset);
+        PAGE_KEYS.put(PAGE_TOTALS_KEY, PaginationKey.totals);
```

Comment:
While you're here (and since it's a good idea) would you mind replacing the other pagination keys with constants?